.dotnet/README.md

```diff
@@ -795,6 +796,52 @@ string message = outputAsJson.RootElement
 
 Notice how you can then call the resulting `ClientResult`'s `GetRawResponse` method and retrieve the response body as `BinaryData` via the `PipelineResponse`'s `Content` property.
 
+### Mock a client for testing
+
+The OpenAI .NET library has been designed to support mocking, providing key features such as:
+- Client methods made virtual to allow overriding.
+- Model Factories to assist in instantiating models that lack public constructors.
+
+To illustrate how mocking works, suppose you want to validate the behavior of the following method using the [Moq](https://github.com/devlooped/moq) library. Given the path to an audio file, it classifies whether it's a safe audio or not based on the presence of the substring "silly":
```

Comment:
Do we want to avoid links to external libraries in the README?

Reply:
I think it should be fine.